nixos/tests/curl-impersonate.nix

```diff
@@ -0,0 +1,124 @@
+import ./make-test-python.nix ({ pkgs, lib, ... }: let
```

Comment:
I was also confused about what this test file was actually testing, so it might be worth adding a comment with some of the information in https://github.com/NixOS/nixpkgs/pull/194310#issuecomment-1272167835

Reply:
Good call! I've added some comments at the top and throughout the test now which hopefully should explain enough of what's going on